[ILM] Refactored policies list to use EuiInMemoryTable

yuliacech opened the pull request:
## Summary

This PR continues work started in https://github.com/elastic/kibana/pull/106243 and https://github.com/elastic/kibana/pull/106734. This is a clean up of the `TableContent` component that now uses `EuiInMemoryTable` instead of a manual rendering of cells and rows of policies. The refactoring resulted in minor UI changes: 
- Actions column displays 2 icon buttons to add a policy to an index template and to delete a policy
#### Before 
![Screenshot 2021-08-03 at 16 48 53](https://user-images.githubusercontent.com/6585477/128047866-8d27a629-1e3b-46bb-8dda-1b897df6cf4e.png)
#### After
<img width="257" alt="Screenshot 2021-08-05 at 13 53 11" src="https://user-images.githubusercontent.com/6585477/128348602-aa55e513-950e-47ff-b88e-433f763271ff.png">

Delete button enabled
<img width="183" alt="Screenshot 2021-08-05 at 13 53 25" src="https://user-images.githubusercontent.com/6585477/128348607-a6dbb5ec-fd97-4084-978e-b3f31cace6b7.png">

Delete button disabled
<img width="308" alt="Screenshot 2021-08-05 at 13 53 33" src="https://user-images.githubusercontent.com/6585477/128348608-53e57f5b-ce30-43ac-b233-54a42a5646e2.png">

- View indices action is now a link in the Linked indices column
<img width="387" alt="Screenshot 2021-08-05 at 13 52 53" src="https://user-images.githubusercontent.com/6585477/128348794-b9c5f78c-6d30-47b1-9ff2-e8ae4d03dbc1.png">


- Pagination now uses default page sizes 
#### Before
![Screenshot 2021-08-03 at 16 49 03](https://user-images.githubusercontent.com/6585477/128048365-f9745c32-2633-4b58-a015-588bdc39de30.png)
#### After
![Screenshot 2021-08-03 at 16 29 37](https://user-images.githubusercontent.com/6585477/128048489-18c26e35-c7ce-4ab8-8bb3-31df7b2efce1.png)
- Version column removed
- Changed modified date format to 'Jul 23, 2021' 
#### Before
![Screenshot 2021-08-03 at 16 48 34](https://user-images.githubusercontent.com/6585477/128048762-8a4e5f7c-e1e8-43f2-9988-44f3ee7a09aa.png)
#### After
<img width="996" alt="Screenshot 2021-08-05 at 13 52 36" src="https://user-images.githubusercontent.com/6585477/128348963-9751032a-196c-4675-8b16-73749721a388.png">

Also fixes https://github.com/elastic/kibana/issues/63189 (mobile view missing column names)
<img width="680" alt="Screenshot 2021-08-05 at 16 28 21" src="https://user-images.githubusercontent.com/6585477/128367464-1e41692b-d35b-40bb-a87f-2841e77f4837.png">



### Checklist

- [x] [Unit or functional tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html) were updated or added to match the most common scenarios
- [x] Any UI touched in this PR is usable by keyboard only (learn more about [keyboard accessibility](https://webaim.org/techniques/keyboard/))
- [x] Any UI touched in this PR does not create any new axe failures (run axe in browser: [FF](https://addons.mozilla.org/en-US/firefox/addon/axe-devtools/), [Chrome](https://chrome.google.com/webstore/detail/axe-web-accessibility-tes/lhdoppojpmngadmnindnejefpokejbdd?hl=en-US))


## Release Note
The Index Lifecycle policies list now has a slightly improved table view. The search bar now also allows use of the query syntax.
<img width="996" alt="Screenshot 2021-08-05 at 13 52 36" src="https://user-images.githubusercontent.com/6585477/128348963-9751032a-196c-4675-8b16-73749721a388.png">

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/142585/)
* Commit: ba3e0e5d9fa85e7a6de15f4cc40ad95a55e46480
* Storybooks not built yet
* This comment will update when the build is complete

### Failed CI Steps
* [Check i18n](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/142585/execution/node/430/log)

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/142585/","number":"142585","commit":"ba3e0e5d9fa85e7a6de15f4cc40ad95a55e46480"}]}
PIPELINE-->

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/142594/)
* Commit: 9441a4a333e55aa7bf7dde133f3490a581a9e854
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-107510/9441a4a333e55aa7bf7dde133f3490a581a9e854)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/ced9aecab517541ca3c941894cc3d60f22845cc6) | [after](https://github.com/elastic/kibana/commit/9441a4a333e55aa7bf7dde133f3490a581a9e854) | diff |
| --- | --- | --- | --- |
| `indexLifecycleManagement` | 213 | 212 | -1 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/ced9aecab517541ca3c941894cc3d60f22845cc6) | [after](https://github.com/elastic/kibana/commit/9441a4a333e55aa7bf7dde133f3490a581a9e854) | diff |
| --- | --- | --- | --- |
| `indexLifecycleManagement` | 249.0KB | 243.4KB | -5.6KB |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/ced9aecab517541ca3c941894cc3d60f22845cc6) | [after](https://github.com/elastic/kibana/commit/9441a4a333e55aa7bf7dde133f3490a581a9e854) | diff |
| --- | --- | --- | --- |
| `indexLifecycleManagement` | 48.6KB | 48.5KB | -81.0B |

### History
* :broken_heart: [Build #142585](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/142585/) failed ba3e0e5d9fa85e7a6de15f4cc40ad95a55e46480

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/142585/","number":"142585","commit":"ba3e0e5d9fa85e7a6de15f4cc40ad95a55e46480"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/142594/","number":"142594","commit":"9441a4a333e55aa7bf7dde133f3490a581a9e854"}]}
PIPELINE-->

elasticmachine commented:
Pinging @elastic/kibana-stack-management (Team:Stack Management)

alisonelizabeth commented on x-pack/plugins/index_lifecycle_management/__jest__/policy_table.test.tsx:
```diff
@@ -169,24 +170,25 @@ describe('policy table', () => {
+    expect(buttons.at(0).text()).toBe('View indices linked to policy');
+    expect(buttons.at(1).text()).toBe('Add policy to index template');
+    expect(buttons.at(2).text()).toBe('Delete policy');
     expect((buttons.at(1).getDOMNode() as HTMLButtonElement).disabled).toBeFalsy();
```
Same question here - is this testing the correct thing? It looks like the view indices button is at index `0` and disabled should be `true`.

```
    expect((buttons.at(0).getDOMNode() as HTMLButtonElement).disabled).toBeTruthy();
```

yuliacech replied:
I updated the button tests, but also plan on working more on test refactoring in a future PR.

alisonelizabeth commented on x-pack/plugins/index_lifecycle_management/__jest__/policy_table.test.tsx:
```diff
@@ -169,24 +170,25 @@ describe('policy table', () => {
     expect(buttons.at(2).text()).toBe('Delete policy');
     expect((buttons.at(2).getDOMNode() as HTMLButtonElement).disabled).toBeTruthy();
```
Is this testing the correct thing? It looks like the view indices button is at index `0` and disabled should be `false`.

```
expect((buttons.at(2).getDOMNode() as HTMLButtonElement).disabled).toBeFalsey();
```

alisonelizabeth commented on x-pack/plugins/index_lifecycle_management/__jest__/policy_table.test.tsx:
```diff
@@ -132,10 +136,10 @@ describe('policy table', () => {
     rendered.update();
-    const fiftyButton = rendered.find('.euiContextMenuItem').at(1);
-    fiftyButton.simulate('click');
+    const numberOfRowsButton = rendered.find('.euiContextMenuItem').at(1);
```
nit: I'm not able to add a comment on the correct line, but I think the test description could be improved. Took me a little bit to understand what this was testing - maybe something like:

```
should show more policies when the "Rows per page" value is increased
```

alisonelizabeth commented on x-pack/plugins/index_lifecycle_management/__jest__/policy_table.test.tsx:
```diff
@@ -132,10 +136,10 @@ describe('policy table', () => {
+    numberOfRowsButton.simulate('click');
     rendered.update();
-    expect(namesText(rendered).length).toBe(50);
+    expect(namesText(rendered).length).toBe(25);
```
nit: I know this was pre-existing, but the naming of `namesText()` isn't very clear.

alisonelizabeth commented on x-pack/plugins/index_lifecycle_management/__jest__/policy_table.test.tsx:
```diff
@@ -146,21 +150,18 @@ describe('policy table', () => {
   });
   test('should sort when modified date header is clicked', () => {
-    testSort('modifiedDate');
+    testSort('modifiedDate_3');
```
nit: are these `data-test-subj` attributes generated automatically by EUI? I found it a little hard to parse initially with the `index` included. If it's possible to pass our own `data-test-subj` I think that might make this more clear.

yuliacech replied:
This I have to leave as is for now, because using `EuiInMemoryTable` prevents us from adding custom `data-test-subj` to the table header buttons.

alisonelizabeth commented on x-pack/plugins/index_lifecycle_management/__jest__/policy_table.test.tsx:
```diff
@@ -197,7 +199,7 @@ describe('policy table', () => {
     const numberOfIndexTemplates = testPolicy.indexTemplates.length;
     expect(firstRow).toBe(
-      `testy0${numberOfIndices}${numberOfIndexTemplates}${testPolicy.version}${testDateFormatted}Actions`
+      `Nametesty0Linked index templates${numberOfIndexTemplates}Linked indices${numberOfIndices}Modified date${testDateFormatted}`
```
I find this a little hard to parse as well 😄 . Is there a better way to test the values?

alisonelizabeth commented on x-pack/plugins/index_lifecycle_management/public/application/sections/policy_table/components/policy_action.tsx:
```diff
@@ -0,0 +1,60 @@
+      <AddPolicyToTemplateConfirmModal
+        policy={policyAction.policy}
+        onSuccess={(indexTemplate: string) => {
+          policyAction!.policy.indexTemplates = [
```
Can you explain the expected behavior here? Shouldn't `setPolicyAction` be used to update the policy?

yuliacech replied:
I added a comment about updating index templates list

alisonelizabeth reviewed (commented):
Nice job refactoring @yuliacech! I think these changes simplify a lot of the code.

I had a hard time following some of the tests in `policy_table.test.tsx`. I understand a lot of these test cases were pre-existing, so I'm open to addressing some of my comments separately.

I also noticed a couple things in the UI. Again, I think unrelated to your changes, but just wanted to point out:

- It might be nice to add a tooltip or something to add more details why a particular action is disabled.
<img width="349" alt="Screen Shot 2021-08-03 at 8 20 21 PM" src="https://user-images.githubusercontent.com/5226211/128103040-f7055eaa-69b2-4a7e-a57b-f7c8aaa646e0.png">
<img width="332" alt="Screen Shot 2021-08-03 at 8 21 55 PM" src="https://user-images.githubusercontent.com/5226211/128103041-5c7296bd-d66c-42aa-92fb-3f696912a673.png">

- I noticed the `ilm-history-ilm-policy` policy says it has a linked index, but if you click on it, the filtered view has no results. I believe this is because data streams are being used.

<img width="953" alt="Screen Shot 2021-08-03 at 8 28 16 PM" src="https://user-images.githubusercontent.com/5226211/128102988-df405810-c248-44c5-b38a-298685dd4528.png">

yuliacech commented:
@elasticmachine merge upstream

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/143206/)
* Commit: c6d82624069d5eb5761db9e11428a9d9a77ac46f
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-107510/c6d82624069d5eb5761db9e11428a9d9a77ac46f)
* This comment will update when the build is complete

### Failed CI Steps
* [Execute xpack-accessibility](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/143206/execution/node/531/log)
* [Execute xpack-accessibility](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/143206/execution/node/793/log)

### History
* :green_heart: [Build #142594](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/142594/) succeeded 9441a4a333e55aa7bf7dde133f3490a581a9e854
* :broken_heart: [Build #142585](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/142585/) failed ba3e0e5d9fa85e7a6de15f4cc40ad95a55e46480

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/142585/","number":"142585","commit":"ba3e0e5d9fa85e7a6de15f4cc40ad95a55e46480"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/142594/","number":"142594","commit":"9441a4a333e55aa7bf7dde133f3490a581a9e854"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/143206/","number":"143206","commit":"c6d82624069d5eb5761db9e11428a9d9a77ac46f"}]}
PIPELINE-->

yuliacech commented:
Thank you so much for your review, @alisonelizabeth! All great points, I think I addressed most of them. 
 
> It might be nice to add a tooltip or something to add more details why a particular action is disabled.

I added a tooltip to the delete button, both when it's enabled and disabled. The 'view linked indices' button is moved to the Linked indices column. 

> I noticed the ilm-history-ilm-policy policy says it has a linked index, but if you click on it, the filtered view has no results. 

This is a known [issue](https://github.com/elastic/kibana/issues/88441), I'll see if I can fix it while working on this project.

yuliacech commented on x-pack/plugins/index_lifecycle_management/public/application/sections/policy_table/policy_table.tsx:
```diff
@@ -5,35 +5,23 @@
  * 2.0.
```
Good suggestion, I renamed the files to `policy_list` and `policy_table` that I think provides a better structure.

yuliacech commented on x-pack/plugins/index_lifecycle_management/public/application/sections/policy_table/policy_table.tsx:
```diff
@@ -139,7 +69,7 @@ export const PolicyTable: React.FunctionComponent<Props> = ({ policies, updatePo
   return (
     <>
-      {confirmModal}
+      <PolicyAction updatePolicies={updatePolicies} />
```
I renamed this to `PolicyListHandler` and the context value to `ListAction` with `selectedPolicy` and `actionType`.

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/143224/)
* Commit: 9e318c71fbb7c752a2dcb0e25043499ff2c4ce80
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-107510/9e318c71fbb7c752a2dcb0e25043499ff2c4ce80)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/9d42d670e1676a622d7ee7d4d564c4d7dc3973e8) | [after](https://github.com/elastic/kibana/commit/9e318c71fbb7c752a2dcb0e25043499ff2c4ce80) | diff |
| --- | --- | --- | --- |
| `indexLifecycleManagement` | 213 | 212 | -1 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/9d42d670e1676a622d7ee7d4d564c4d7dc3973e8) | [after](https://github.com/elastic/kibana/commit/9e318c71fbb7c752a2dcb0e25043499ff2c4ce80) | diff |
| --- | --- | --- | --- |
| `indexLifecycleManagement` | 249.0KB | 244.6KB | -4.4KB |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/9d42d670e1676a622d7ee7d4d564c4d7dc3973e8) | [after](https://github.com/elastic/kibana/commit/9e318c71fbb7c752a2dcb0e25043499ff2c4ce80) | diff |
| --- | --- | --- | --- |
| `indexLifecycleManagement` | 48.6KB | 48.5KB | -81.0B |

### History
* :broken_heart: [Build #143206](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/143206/) failed c6d82624069d5eb5761db9e11428a9d9a77ac46f
* :green_heart: [Build #142594](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/142594/) succeeded 9441a4a333e55aa7bf7dde133f3490a581a9e854
* :broken_heart: [Build #142585](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/142585/) failed ba3e0e5d9fa85e7a6de15f4cc40ad95a55e46480

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/142585/","number":"142585","commit":"ba3e0e5d9fa85e7a6de15f4cc40ad95a55e46480"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/142594/","number":"142594","commit":"9441a4a333e55aa7bf7dde133f3490a581a9e854"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/143206/","number":"143206","commit":"c6d82624069d5eb5761db9e11428a9d9a77ac46f"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/143224/","number":"143224","commit":"9e318c71fbb7c752a2dcb0e25043499ff2c4ce80"}]}
PIPELINE-->

alisonelizabeth reviewed (approved):
Thanks for addressing my feedback @yuliacech! Did not retest, but reviewed your latest changes and code LGTM.

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/143243/)
* Commit: d06c155d9fef53ed682823592f95ac33ce4b5b42
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-107510/d06c155d9fef53ed682823592f95ac33ce4b5b42)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/9d42d670e1676a622d7ee7d4d564c4d7dc3973e8) | [after](https://github.com/elastic/kibana/commit/d06c155d9fef53ed682823592f95ac33ce4b5b42) | diff |
| --- | --- | --- | --- |
| `indexLifecycleManagement` | 213 | 212 | -1 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/9d42d670e1676a622d7ee7d4d564c4d7dc3973e8) | [after](https://github.com/elastic/kibana/commit/d06c155d9fef53ed682823592f95ac33ce4b5b42) | diff |
| --- | --- | --- | --- |
| `indexLifecycleManagement` | 249.0KB | 244.6KB | -4.4KB |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/9d42d670e1676a622d7ee7d4d564c4d7dc3973e8) | [after](https://github.com/elastic/kibana/commit/d06c155d9fef53ed682823592f95ac33ce4b5b42) | diff |
| --- | --- | --- | --- |
| `indexLifecycleManagement` | 48.6KB | 48.5KB | -81.0B |

### History
* :green_heart: [Build #143224](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/143224/) succeeded 9e318c71fbb7c752a2dcb0e25043499ff2c4ce80
* :broken_heart: [Build #143206](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/143206/) failed c6d82624069d5eb5761db9e11428a9d9a77ac46f
* :green_heart: [Build #142594](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/142594/) succeeded 9441a4a333e55aa7bf7dde133f3490a581a9e854
* :broken_heart: [Build #142585](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/142585/) failed ba3e0e5d9fa85e7a6de15f4cc40ad95a55e46480

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/142585/","number":"142585","commit":"ba3e0e5d9fa85e7a6de15f4cc40ad95a55e46480"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/142594/","number":"142594","commit":"9441a4a333e55aa7bf7dde133f3490a581a9e854"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/143206/","number":"143206","commit":"c6d82624069d5eb5761db9e11428a9d9a77ac46f"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/143224/","number":"143224","commit":"9e318c71fbb7c752a2dcb0e25043499ff2c4ce80"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/143243/","number":"143243","commit":"d06c155d9fef53ed682823592f95ac33ce4b5b42"}]}
PIPELINE-->

yuliacech commented:
@elasticmachine merge upstream

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/143592/)
* Commit: b3afe7fca2eb98fdfe1bee03157d50f8d3e819aa
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-107510/b3afe7fca2eb98fdfe1bee03157d50f8d3e819aa)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/78cc089ec2426a2c4c9d3e29ce05c5ed435f431f) | [after](https://github.com/elastic/kibana/commit/b3afe7fca2eb98fdfe1bee03157d50f8d3e819aa) | diff |
| --- | --- | --- | --- |
| `indexLifecycleManagement` | 213 | 212 | -1 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/78cc089ec2426a2c4c9d3e29ce05c5ed435f431f) | [after](https://github.com/elastic/kibana/commit/b3afe7fca2eb98fdfe1bee03157d50f8d3e819aa) | diff |
| --- | --- | --- | --- |
| `indexLifecycleManagement` | 249.0KB | 244.6KB | -4.4KB |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/78cc089ec2426a2c4c9d3e29ce05c5ed435f431f) | [after](https://github.com/elastic/kibana/commit/b3afe7fca2eb98fdfe1bee03157d50f8d3e819aa) | diff |
| --- | --- | --- | --- |
| `indexLifecycleManagement` | 48.6KB | 48.5KB | -81.0B |

### History
* :green_heart: [Build #143243](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/143243/) succeeded d06c155d9fef53ed682823592f95ac33ce4b5b42
* :green_heart: [Build #143224](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/143224/) succeeded 9e318c71fbb7c752a2dcb0e25043499ff2c4ce80
* :broken_heart: [Build #143206](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/143206/) failed c6d82624069d5eb5761db9e11428a9d9a77ac46f
* :green_heart: [Build #142594](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/142594/) succeeded 9441a4a333e55aa7bf7dde133f3490a581a9e854
* :broken_heart: [Build #142585](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/142585/) failed ba3e0e5d9fa85e7a6de15f4cc40ad95a55e46480

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/142585/","number":"142585","commit":"ba3e0e5d9fa85e7a6de15f4cc40ad95a55e46480"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/142594/","number":"142594","commit":"9441a4a333e55aa7bf7dde133f3490a581a9e854"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/143206/","number":"143206","commit":"c6d82624069d5eb5761db9e11428a9d9a77ac46f"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/143224/","number":"143224","commit":"9e318c71fbb7c752a2dcb0e25043499ff2c4ce80"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/143243/","number":"143243","commit":"d06c155d9fef53ed682823592f95ac33ce4b5b42"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/143592/","number":"143592","commit":"b3afe7fca2eb98fdfe1bee03157d50f8d3e819aa"}]}
PIPELINE-->

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/144902/)
* Commit: b3afe7fca2eb98fdfe1bee03157d50f8d3e819aa
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-107510/b3afe7fca2eb98fdfe1bee03157d50f8d3e819aa)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/78cc089ec2426a2c4c9d3e29ce05c5ed435f431f) | [after](https://github.com/elastic/kibana/commit/b3afe7fca2eb98fdfe1bee03157d50f8d3e819aa) | diff |
| --- | --- | --- | --- |
| `indexLifecycleManagement` | 213 | 212 | -1 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/78cc089ec2426a2c4c9d3e29ce05c5ed435f431f) | [after](https://github.com/elastic/kibana/commit/b3afe7fca2eb98fdfe1bee03157d50f8d3e819aa) | diff |
| --- | --- | --- | --- |
| `indexLifecycleManagement` | 249.0KB | 244.6KB | -4.4KB |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/78cc089ec2426a2c4c9d3e29ce05c5ed435f431f) | [after](https://github.com/elastic/kibana/commit/b3afe7fca2eb98fdfe1bee03157d50f8d3e819aa) | diff |
| --- | --- | --- | --- |
| `indexLifecycleManagement` | 48.6KB | 48.5KB | -81.0B |

### History
* :green_heart: [Build #143592](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/143592/) succeeded b3afe7fca2eb98fdfe1bee03157d50f8d3e819aa
* :green_heart: [Build #143243](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/143243/) succeeded d06c155d9fef53ed682823592f95ac33ce4b5b42
* :green_heart: [Build #143224](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/143224/) succeeded 9e318c71fbb7c752a2dcb0e25043499ff2c4ce80
* :broken_heart: [Build #143206](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/143206/) failed c6d82624069d5eb5761db9e11428a9d9a77ac46f
* :green_heart: [Build #142594](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/142594/) succeeded 9441a4a333e55aa7bf7dde133f3490a581a9e854

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/142594/","number":"142594","commit":"9441a4a333e55aa7bf7dde133f3490a581a9e854"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/143206/","number":"143206","commit":"c6d82624069d5eb5761db9e11428a9d9a77ac46f"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/143224/","number":"143224","commit":"9e318c71fbb7c752a2dcb0e25043499ff2c4ce80"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/143243/","number":"143243","commit":"d06c155d9fef53ed682823592f95ac33ce4b5b42"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/143592/","number":"143592","commit":"b3afe7fca2eb98fdfe1bee03157d50f8d3e819aa"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/144902/","number":"144902","commit":"b3afe7fca2eb98fdfe1bee03157d50f8d3e819aa"}]}
PIPELINE-->
